script/validate_builds.py

```diff
@@ -48,19 +50,51 @@ def trigger_basic_validation_checks(file_name):
                     elif key in line:
                         matched_keys.append(key)
                         package_data[key_checks[key]] = line.split(':')[-1].strip()
-            except IndexError as ie:
-                raise IndexError(str(ie))
+                        if line.startswith('# Travis-Check'):
+                            key_value = line.split(':')
+                            should_travis_run_for_script = eval(key_value[-1].strip())
+                        elif line.startswith('# Package'):
+                            key_value = line.split(':')
+                            package_name = key_value[1].strip()
+                        elif line.startswith('# Version'):
+                            key_value = line.split(':')
+                            package_version = key_value[1].strip()
+                        elif line.startswith('# Source repo'):
+                            key_value = line.split(':')
+                            package_url = ":".join(key_value[1:]).strip()
+            except (IndexError, TypeError) as ie:
+                print("Failure: {}".format(str(ie)))
+                raise Exception(str(ie))
+            # Assuming, actual script will be available, once doc string is completely parsed.
+            if package_name and 'PACKAGE_NAME=' in line:
+                key_val = line.split('=')
+                alt_package_name_value = "${1:-%s}"%package_name
+                if str(package_name) != str(key_val[-1].strip()) and str(alt_package_name_value) != str(key_val[-1].strip()):
+                    raise ValueError("PACKAGE_NAME in script, does not match the Package in doc string")
+            elif package_version and 'PACKAGE_VERSION=' in line:
+                key_val = line.split('=')
+                alt_version_value = "${1:-%s}"%package_version
+                if str(package_version) != str(key_val[-1].strip()) and str(alt_version_value) != str(key_val[-1].strip()):
+                    raise ValueError("PACKAGE_VERSION in script, does not match the Version in doc string")
```

Comment:
There is a chance that, developer can commit a build-script for multiple versions.
In that case,  below is the string:
`# Version : version01, version02`

While setting the package version, it is possible for having a single version as Default version, 
`PACKAGE_VERSION=${1:-version01}`.

Above check will fail at this case.

As we are mandating the package version is configurable, we allow only, `PACKAGE_VERSION=${1:-version01}` format.

so can we have condition something like:
```
if package_version and ('PACKAGE_VERSION=' in line or 'VERSION=' in line):
    pattern = re.compile("(PACKAGE_VERSION|VERSION)=\${\d*:-(?P<version>[^}]*?)}")
    match = pattern.match(line)
    default_version = match.group("version")
    if default_version:
        if default_version not in [x.strip() for x in package_version.split (',')]:
            raise ValueError("PACKAGE_VERSION/VERSION in script, does not match the Version in doc string")
    else:
        raise ValueError("PACKAGE_VERSION/VERSION in script, does not have correct format")
```

CC: @seth-priya , Please correct me if I am wrong.